nifi-nar-bundles/nifi-splunk-bundle/nifi-splunk-processors/src/main/java/org/apache/nifi/processors/splunk/QuerySplunkIndexingStatus.java

```diff
@@ -0,0 +1,237 @@
+            .required(true)
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor MAX_QUERY_SIZE = new PropertyDescriptor.Builder()
+            .name("max-query-size")
+            .displayName("Maximum Query Size")
+            .description(
+                    "The maximum number of acknowledgement identifiers the outgoing query contains in one batch. " +
+                    "It is recommended not to set it too low in order to reduce network communication.")
+            .defaultValue("10000")
+            .required(true)
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .build();
+
+    private volatile Integer maxQuerySize;
+    private volatile Integer ttl;
+
+    @Override
+    public List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> result = new ArrayList<>();
+        final List<PropertyDescriptor> common = super.getSupportedPropertyDescriptors();
+        result.addAll(common);
+        result.add(TTL);
+        result.add(MAX_QUERY_SIZE);
+        return result;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        super.onScheduled(context);
+        maxQuerySize = context.getProperty(MAX_QUERY_SIZE).asInteger();
+        ttl = context.getProperty(TTL).asTimePeriod(TimeUnit.MILLISECONDS).intValue();
+    }
+
+    @OnStopped
+    public void onUnscheduled() {
+        super.onUnscheduled();
+        maxQuerySize = null;
+        ttl = null;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        final RequestMessage requestMessage;
+        final List<FlowFile> flowFiles = session.get(maxQuerySize);
+
+        if (flowFiles.isEmpty()) {
+            return;
+        }
+
+        final long currentTime = System.currentTimeMillis();
+        final Map<Long, FlowFile> undetermined = new HashMap<>();
+
+        for (final FlowFile flowFile : flowFiles)  {
+            final Optional<Long> sentAt = extractLong(flowFile.getAttribute(SplunkAPICall.RESPONDED_AT_ATTRIBUTE));
+            final Optional<Long> ackId = extractLong(flowFile.getAttribute(SplunkAPICall.ACKNOWLEDGEMENT_ID_ATTRIBUTE));
+
+            if (!sentAt.isPresent() || !ackId.isPresent()) {
+                getLogger().error("Flow file ({}) attributes {} and {} are expected to be set using 64-bit integer values!",
+                        new Object[]{flowFile.getId(), SplunkAPICall.RESPONDED_AT_ATTRIBUTE, SplunkAPICall.ACKNOWLEDGEMENT_ID_ATTRIBUTE});
+                session.transfer(flowFile, RELATIONSHIP_FAILURE);
+            } else if (sentAt.get() + ttl < currentTime) {
+                session.transfer(flowFile, RELATIONSHIP_UNACKNOWLEDGED);
```

Comment:
There is a risk for that, given it is sitting enough time in the queue, but there must be an upper limit: Splunk has no explicit "unacknowledged" answer, thus it will return with a false value for cases where it is timed out in Splunk as well (or non-existing, or troubled ones or for ones with a previous positive answer. The default value of the TTL is 1 hour, which I believe should be enough for most healthy environment.

I could imagine an attribute on the flow file (something like a counter for attempts or even just a flag) would help. Are you okay with adding this in a follow up ticket?

Reply:
Yeah I was thinking along the same lines - adding some sort of flag to indicate that at least one attempt had been made. That can probably be done in a follow-on ticket, that's fine.

Reply:
https://issues.apache.org/jira/browse/NIFI-8176